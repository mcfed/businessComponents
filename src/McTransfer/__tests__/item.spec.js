import React from 'react';
import {shallow} from 'enzyme';
import Item from '../ts/item.tsx';

function setup(props, options) {
  const defaultProps = {
    item: {
      disabled: false
    },
    renderedText: '',
    renderedEl: '',
    style: {},
    checked: true,
    onClick: jest.fn(),
    prefixCls: ''
  };
  const wrapper = shallow(
    <Item {...Object.assign({}, defaultProps, props)} />,
    options
  );

  return {
    wrapper
  };
}

describe('快照测试', () => {
  it('全页快照', () => {
    const props = {
      item: {
        disabled: false
      }
    };
    const {wrapper} = setup(props);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('方法测试全覆盖', () => {
  const props = {
    item: {
      disabled: true
    }
  };
  it('shouldComponentUpdate 方法测试', () => {
    const {wrapper} = setup(props);
    const instance = wrapper.instance();
    instance.shouldComponentUpdate();
  });
  it('li onClick 方法测试', () => {
    props.item.disabled = false;
    const {wrapper} = setup(props);
    wrapper
      .find('li')
      .props()
      .onClick();
  });
});
