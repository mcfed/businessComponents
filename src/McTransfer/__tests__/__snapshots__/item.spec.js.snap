// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 全页快照 1`] = `
<li
  className="-content-item"
  onClick={[Function]}
  style={Object {}}
  title=""
>
  <Checkbox
    checked={true}
    disabled={false}
    indeterminate={false}
  />
  <span />
</li>
`;
